apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -44,6 +66,63 @@ export const askAITutor = createAsyncThunk(
   }
 );
 
+// General Chat
+// This thunk's callback function submits a user chat message to the chat completion endpoint,
+// waits for a chat completion response, and updates the user message state.
+export const submitChatMessage = createAsyncThunk(
+  'aitutor/submitChatMessage',
+  async (message: string, thunkAPI) => {
+    const state = thunkAPI.getState() as {aiTutor: AITutorState};
+    const systemPrompt =
+      'You are an assistant teacher in a high school classroom where the students are learning Java using the Code.org curriculum. Answer their questions in plain, easy-to-understanding English. Do not write any code in your response.';
+
+    const storedMessages = state.aiTutor.chatMessages;
+    const newMessageId =
+      storedMessages.length === 0
+        ? 1
+        : storedMessages[storedMessages.length - 1].id + 1;
+    const appropriateChatMessages = storedMessages.filter(
+      msg => msg.status === Status.OK
+    );
+
+    // Create the new user ChatCompleteMessage and add to chatMessages.
+    const newMessage: ChatCompletionMessage = {
+      id: newMessageId,
+      role: Role.USER,
+      status: Status.UNKNOWN,
+      chatMessageText: message,
+    };
+    thunkAPI.dispatch(addChatMessage(newMessage));
+
+    // Send user message to backend and retrieve assistant response.
+    const chatApiResponse = await getChatCompletionMessage(
+      systemPrompt,
+      newMessageId,
+      message,
+      appropriateChatMessages
+    );
+
+    // Find message in chatMessages and update status.
+    thunkAPI.dispatch(
+      updateChatMessageStatus({
+        id: chatApiResponse.id,
+        status: chatApiResponse.status,
+      })
+    );
+
+    // Add assistant chat messages to chatMessages.
+    if (chatApiResponse.assistantResponse) {
+      const assistantChatMessage: ChatCompletionMessage = {
+        id: chatApiResponse.id + 1,
```

Comment:
What is the role of the id? I'm wondering if we could remove them and avoid having to update/maintain these values.